tests/src/Integration/Action/UserUnblockTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\UserUnblockTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\UserUnblock
+ * @group rules_actions
+ */
+class UserUnblockTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * Constant used for authenticated test when mocking a user.
+   */
+  const AUTHENTICATED = TRUE;
+
+  /**
+   * Constant used for authenticated test when mocking a user.
+   */
+  const ANONYMOUS = FALSE;
+
+  /**
+   * Constant used for active test when mocking a user.
+   */
+  const ACTIVE = TRUE;
+
+  /**
+   * Constant used for active test when mocking a user.
+   */
+  const BLOCKED = FALSE;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Engine\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * Test execute() method for users with different status.
+   * @dataProvider userProvider
+   * @covers ::execute
+   */
+  public function testUnblockUser($active, $authenticated, $expects) {
+    $user = $this->getMock('Drupal\user\UserInterface');
+
+    $user->expects($this->any())
+      ->method('isBlocked')
+      ->willReturn(!$active);
+
+    $user->expects($this->any())
+      ->method('isAuthenticated')
+      ->willReturn($authenticated);
+
+    $user->expects($this->{$expects}())
+      ->method('activate');
+
+    $this->action->setContextValue('user', $user);
+
+    $this->action->execute();
+  }
```

Comment:
After that we should call autoSaveContext() on the action and make sure it returns the correct user context key or not.